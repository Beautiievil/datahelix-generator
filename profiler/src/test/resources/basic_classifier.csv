date,name,country,currency
10-10-2018,Someone,UK,GBP
2019-05-06 10:18:000,Someone else,AR,USD