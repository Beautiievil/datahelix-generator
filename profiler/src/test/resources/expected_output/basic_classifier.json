{
  "schemaVersion" : "v3",
  "fields" : [ {
    "name" : "id"
  }, {
    "name" : "last_name"
  }, {
    "name" : "email"
  }, {
    "name" : "gender"
  }, {
    "name" : "ip_address"
  }, {
    "name" : "isAdmin"
  }, {
    "name" : "country"
  }, {
    "name" : "when"
  }, {
    "name" : "someNumber"
  }, {
    "name" : "someIntNumber"
  }, {
    "name" : "stockSymbol"
  }, {
    "name" : "title"
  }, {
    "name" : "currency"
  }, {
    "name" : "randomDate"
  } ],
  "rules" : [ {
    "rule" : "Numeric field id rule",
    "constraints" : [ {
      "field" : "id",
      "is" : "ofType",
      "value" : "numeric"
    }, {
      "field" : "id",
      "is" : "greaterThanOrEqualTo",
      "value" : "1"
    }, {
      "field" : "id",
      "is" : "lessThan",
      "value" : "1000"
    }, {
      "field" : "id",
      "is" : "hasDecimalPlaces",
      "value" : 0
    } ]
  }, {
    "rule" : "String field last_name rule",
    "constraints" : [ {
      "field" : "last_name",
      "is" : "ofType",
      "value" : "string"
    }, {
      "field" : "last_name",
      "is" : "matchingRegex",
      "value" : ".{3,15}"
    } ]
  }, {
    "rule" : "String field email rule",
    "constraints" : [ {
      "field" : "email",
      "is" : "ofType",
      "value" : "string"
    }, {
      "field" : "email",
      "is" : "matchingRegex",
      "value" : "(tim|frank|greg)\\.(smith|jones|sestero)@(gmail|hotmail|live)\\.com"
    } ]
  }, {
    "rule" : "String field gender rule",
    "constraints" : [ {
      "field" : "gender",
      "is" : "ofType",
      "value" : "string"
    }, {
      "field" : "gender",
      "is" : "matchingRegex",
      "value" : ".{4,6}"
    } ]
  }, {
    "rule" : "String field ip_address rule",
    "constraints" : [ {
      "field" : "ip_address",
      "is" : "ofType",
      "value" : "string"
    }, {
      "field" : "ip_address",
      "is" : "matchingRegex",
      "value" : ".{9,15}"
    } ]
  }, {
    "rule" : "String field isAdmin rule",
    "constraints" : [ {
      "field" : "isAdmin",
      "is" : "ofType",
      "value" : "string"
    }, {
      "field" : "isAdmin",
      "is" : "matchingRegex",
      "value" : ".{4,5}"
    } ]
  }, {
    "rule" : "String field country rule",
    "constraints" : [ {
      "field" : "country",
      "is" : "ofType",
      "value" : "string"
    }, {
      "field" : "country",
      "is" : "matchingRegex",
      "value" : ".{2,2}"
    } ]
  }, {
    "rule" : "String field when rule",
    "constraints" : [ {
      "field" : "when",
      "is" : "ofType",
      "value" : "string"
    }, {
      "field" : "when",
      "is" : "matchingRegex",
      "value" : ".{4,7}"
    } ]
  }, {
    "rule" : "Numeric field someNumber rule",
    "constraints" : [ {
      "field" : "someNumber",
      "is" : "ofType",
      "value" : "numeric"
    }, {
      "field" : "someNumber",
      "is" : "greaterThanOrEqualTo",
      "value" : "2.32"
    }, {
      "field" : "someNumber",
      "is" : "lessThan",
      "value" : "996.3"
    } ]
  }, {
    "rule" : "Numeric field someIntNumber rule",
    "constraints" : [ {
      "field" : "someIntNumber",
      "is" : "ofType",
      "value" : "numeric"
    }, {
      "field" : "someIntNumber",
      "is" : "greaterThanOrEqualTo",
      "value" : "2"
    }, {
      "field" : "someIntNumber",
      "is" : "lessThan",
      "value" : "1000"
    }, {
      "field" : "someIntNumber",
      "is" : "hasDecimalPlaces",
      "value" : 0
    } ]
  }, {
    "rule" : "String field stockSymbol rule",
    "constraints" : [ {
      "field" : "stockSymbol",
      "is" : "ofType",
      "value" : "string"
    }, {
      "field" : "stockSymbol",
      "is" : "matchingRegex",
      "value" : ".{1,14}"
    } ]
  }, {
    "rule" : "String field title rule",
    "constraints" : [ {
      "field" : "title",
      "is" : "ofType",
      "value" : "string"
    }, {
      "field" : "title",
      "is" : "matchingRegex",
      "value" : ".{2,9}"
    } ]
  }, {
    "rule" : "String field currency rule",
    "constraints" : [ {
      "field" : "currency",
      "is" : "ofType",
      "value" : "string"
    }, {
      "field" : "currency",
      "is" : "matchingRegex",
      "value" : ".{3,3}"
    } ]
  }, {
    "rule" : "randomDate",
    "constraints" : [ {
      "field" : "randomDate",
      "is" : "ofType",
      "value" : "temporal"
    }, {
      "field" : "randomDate",
      "is" : "greaterThanOrEqualTo",
      "value" : "2017-08-11 00:00:00.0"
    }, {
      "field" : "randomDate",
      "is" : "lessThan",
      "value" : "2019-09-14 00:00:00.0"
    } ]
  } ]
}